app/javascript/state/gdpr/index.js

```diff
@@ -0,0 +1,91 @@
+// @flow
+
+export type ConsentState = {
+  previouslyConsented: boolean,
+  consented: ?boolean,
+  countryCode: ?string,
+  email: ?string,
+  isEU: false,
+  memberId: ?string,
+  variant: 'simple' | 'selectable-buttons' | 'scrolling',
```

Comment:
Mainly to avoid having props being passed from various places. Since most props are already in the `consent` state, it made sense to put all props there and make it a connected component. The reason I'm using a redux store in the first place was so that I can read the values from outside the component (e.g. I'm reading country, isEU, etc in the third variant so that I can hide/show elements with jQuery depending on the state of the component).